File: src/geometryService/cpp/stractNet/WayPointFrag.cpp
///////////////////////////////////////////////////////////
//  WayPointFrag.cpp
//  Implementation of the Class WayPointFrag
//  Created on:      06-Aug-2008 7:52:49 AM
//  Original author: david.h.loman
///////////////////////////////////////////////////////////

#include "WayPointFrag.h"
///////////////////////////////////////////////////////////////////////
/*
 *
 * xSTRUCTORS
 *
 */
///////////////////////////////////////////////////////////////////////
WayPointFrag::~WayPointFrag(){}
WayPointFrag::WayPointFrag(WayPoint& way){
	MsgFrag("WayPointFrag", way, MsgFrag::mftWaypoint);
}
WayPointFrag::WayPointFrag(DataInputStream& in){
	MsgFrag("WayPointFrag", in, MsgFrag::mftWaypoint);
}
///////////////////////////////////////////////////////////////////////
/*
 *
 * METHODS
 *
 */
///////////////////////////////////////////////////////////////////////
void WayPointFrag::Deserialize(DataInputStream& in){
	try {
		//Read WayPoint
		Waypoint wp(in); //TODO Check to make sure this is the right way to declare a class!!!!
		this->field = wp;

	} catch (IOException ioe) {
		SNRoot::err("IOException in Deserialize()", 0);
	}
	return;
}
void WayPointFrag::Serialize_Specific(DataOutputStream& out){
	try {
		// write in new data
		out.write(this->field.Serialize());
	} catch (IOException ioe) {
		SNRoot::err("IOException in Serialize()", 0);
	}
	return;
}

/**
 * Custom Getter
 */
WayPoint& WayPointFrag::getWayPoint(){
	return (WayPoint)this->getField();
}

File: src/superceded_GS/java/stractNet/src/stractNet/messaging/msgs/frags/WayPointFrag.java
package stractNet.messaging.msgs.frags;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import stractNet.messaging.WayPoint;

public class WayPointFrag extends MsgFrag<WayPoint> {

	public WayPointFrag(WayPoint wp) {
		super("WayPointFrag", wp, MsgFrag.mftWaypoint);
	}
	
	
	public WayPointFrag(DataInputStream in) {
		super("WayPointFrag", in, MsgFrag.mftWaypoint);
	}

	@Override
	protected void Deserialize(DataInputStream in) {
		try {
			//Read WayPoint
			this.field = new WayPoint(in);
		} catch (IOException ioe) {
			this.err("IOException in Deserialize()", 0);
		}
		return;
	}

	@Override
	protected void Serialize_Specific(DataOutputStream out) {
		try {
			// write in new data
			out.write(this.field.Serialize());
		} catch (IOException ioe) {
			this.err("IOException in Serialize()", 0);
		}

		return;
	}


	//Custom Getter
	public WayPoint getWayPoint() {
		return (WayPoint)this.getField();
	}
	
}
